File: steps/login.steps.js
const { Given, When, Then } = require('@cucumber/cucumber');
const { LoginPage } = require('../pages/LoginPage');

Given('کاربر صفحه ورود دیجی‌پی را باز کرده است', async function () {
  this.loginPage = new LoginPage(this.page);
  await this.loginPage.open();
});

When('شماره موبایل معتبر خود را وارد می‌کند', async function () {
  await this.loginPage.submitMobile(this.env.testMobile);
});

When('شماره موبایل نامعتبر وارد می‌کند', async function () {
  await this.loginPage.submitMobile(this.env.invalidMobile);
});

Then('باید وارد مرحله دریافت کد تایید شود', async function () {
  await this.loginPage.expectOtpStep();
});

Then('باید پیام خطای اعتبارسنجی شماره موبایل را ببیند', async function () {
  await this.loginPage.expectMobileValidation();
});

File: tests/steps/login_steps.py
"""pytest-bdd step definitions for login scenarios."""

from pytest_bdd import given, then, when
from data.test_data import TestData
from pages.login_page import LoginPage
from utils.config import settings


@given("the user is on the login page")
def open_login_page(page):
    login_page = LoginPage(page)
    login_page.open_login()
    return login_page


@when("the user submits a valid phone number")
def submit_valid_phone(open_login_page: LoginPage):
    open_login_page.login_with_phone(settings.phone_number)


@when("the user submits an invalid phone number")
def submit_invalid_phone(open_login_page: LoginPage):
    invalid_phone = TestData.login(settings.phone_number).invalid_phone
    open_login_page.login_with_phone(invalid_phone)


@then("the OTP step should be displayed")
def otp_step_should_be_displayed(open_login_page: LoginPage):
    open_login_page.expect_otp_step()


@then("a login validation error should be displayed")
def login_validation_error_should_be_displayed(open_login_page: LoginPage):
    open_login_page.expect_validation_error()
